### PR TITLE
M2P-82 Further JS refactoring

### PR DESCRIPTION
**Pull request by vitaliyreznikov**
In this PR:
- Remove legacy code we added for support multistep&paymentonly on the same page (we don't use this option for years)
- Mark code that we need only for payment only (wrap up this code into if statement). I intentionally don't format indents for PR readability. Will format in separate PR
- Remove unnecessary DOM listener - we no need to listen to "sum" changes anymore.

Fixes: [M2P-82]

#changelog M2P-82 Further JS refactoring

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactoring

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: view/frontend/templates/js/replacejs.phtml [hunk 1]
```diff
@@ -587,7 +587,7 @@ ob_start();
             }
         };
 
-        ////////////////////////////////////
+        if (getCheckoutType()=="payment") {
```
Comment (danac-gs) [inline view/frontend/templates/js/replacejs.phtml:590]: Pre-fetch should be called on multi step checkout, not payment only.

File: view/frontend/templates/js/replacejs.phtml [hunk 2]
```diff
@@ -1017,6 +994,7 @@ ob_start();
         };
         ///////////////////////////////////////////////////////////
 
+        if (getCheckoutType()=="payment") {
```
Comment (danac-gs) [inline view/frontend/templates/js/replacejs.phtml:997]: Same here.

File: view/frontend/templates/js/replacejs.phtml (hunk 1 above)
Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml:590]: How I can emulate this mode on my local site?

File: view/frontend/templates/js/replacejs.phtml (hunk 2 above)
Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml:997]: The same question. How can I emulate this?

File: view/frontend/templates/js/replacejs.phtml (hunk 1 above)
Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml:590]: @vitaliyreznikov do we support payment only checkout? Was it ever used? I thought we should remove it.

Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml:590]: I asked @Abhinay-bolt and @usmanismail-bolt and as I understand we decided to save payment only for now.

About pre-fetch. Do I understand correctly we use it only to save shipping calculation in cache and get it from cache on shipping&tax call?

File: view/frontend/templates/js/replacejs.phtml (hunk 2 above)
Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml:999]: Do I understand correctly? In multistep we want to use only three fields for hints: country, State/Province and Zip code?

Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml:999]: image

Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml:999]: @vitaliyreznikov those 3 fields are enough to get the shipping pre-fetch / estimation. In many cases the actual street address does not make a difference. Even the state is not mandatory, not all countries have states or regions. So yes, we call prefetch with whatever data we have available and save the results in cache. However, when prefetch is triggered it does a few more calculations if possible, based on the following
1. address saved to quote, if any
2. logged in customer default address, if any
3. address determined by geolocation service (https://ipstack.com/), if configured.

Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml]: Ok, so let me don't change it for now. Updated.

File: view/frontend/templates/js/replacejs.phtml (hunk 1 above)
Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml]: Fixed

File: view/frontend/templates/js/replacejs.phtml [hunk 3]
```diff
@@ -800,6 +794,8 @@ ob_start();
                             allowAutoOpen = false;
                         }
                     });
+            // prefetch Shipping and Tax for multi-step checkout
+            prefetchShipping();
```
Comment (mccottry) [inline view/frontend/templates/js/replacejs.phtml:798]: This makes prefetch happen unnecessarily often.
consider bundling this with with the `boltcart` customer section, or
create a new `bolthints` customer section have it invalidate on `/V1/carts/`*/estimate-shipping-methods` and `/V1/guest-carts/*/estimate-shipping-methods`

Comment (vitaliyreznikov) [reply view/frontend/templates/js/replacejs.phtml]: Good catch Leon! I fixed this by changing "POST" to "GET" in prefetchShipping function. We no need to invalidate any sections when we do this prefetch call.

And this is an interesting idea to handle prefetchShipping using sections down the road, let me think about it.

**Comment by vitaliyreznikov** [comment]
@mccottry Are we ok to merge this one?

File: view/frontend/templates/js/replacejs.phtml (hunk 3 above)
Comment (mccottry) [reply view/frontend/templates/js/replacejs.phtml]: sounds good.

**Pull request merged**